airflow/providers/google/cloud/sensors/dataflow.py

```diff
@@ -113,3 +113,75 @@ def poke(self, context: dict) -> bool:
             raise AirflowException(f"Job with id '{self.job_id}' is already in terminal state: {job_status}")
 
         return False
+
+
+class DataflowJobMetricsSensor(BaseSensorOperator):
+    """
+    Checks for the status of a job in Google Dataflow.
+
+    :param job_id: ID of the job to be checked.
+    :type job_id: str
+    :param callback: callback which is called with list of read job metrics
+        See:
+        https://cloud.google.com/dataflow/docs/reference/rest/v1b3/MetricUpdate
+    :type callback: callable
+    :param project_id: Optional, the Google Cloud project ID in which to start a job.
+        If set to None or missing, the default project_id from the Google Cloud connection is used.
+    :type project_id: str
+    :param location: Job location.
+    :type location: str
+    :param gcp_conn_id: The connection ID to use connecting to Google Cloud.
+    :type gcp_conn_id: str
+    :param delegate_to: The account to impersonate using domain-wide delegation of authority,
+        if any. For this to work, the service account making the request must have
+        domain-wide delegation enabled.
+    :type delegate_to: str
+    :param impersonation_chain: Optional service account to impersonate using short-term
+        credentials, or chained list of accounts required to get the access_token
+        of the last account in the list, which will be impersonated in the request.
+        If set as a string, the account must grant the originating account
+        the Service Account Token Creator IAM role.
+        If set as a sequence, the identities from the list must grant
+        Service Account Token Creator IAM role to the directly preceding identity, with first
+        account from the list granting this role to the originating account (templated).
+    :type impersonation_chain: Union[str, Sequence[str]]
+    """
+
+    template_fields = ['job_id']
+
+    @apply_defaults
+    def __init__(
+        self,
+        *,
+        job_id: str,
+        callback: Callable,
```

Comment:
Have you check how well it works with serialisation? Serialising a function can be problematic.cc @kaxil @mik-laj

Reply:
@turbaszek could you provide some more information how to test it for serialization, please? I am not so familiar with this topic

Reply:
it is safe for serialization.